doc/guides/pm/device_runtime.rst

```diff
@@ -142,20 +142,28 @@ special synchronization is required.
 
 To enable device runtime power management on a device, the driver needs to call
 :c:func:`pm_device_runtime_enable` at initialization time. Note that this
-function will put the device into the :c:enumerator:`PM_DEVICE_STATE_SUSPENDED`
-state. In case this does not reflect the actual device state, the init function
-must perform the necessary operations to suspend the device.
+function will suspend the device if its state is
+:c:enumerator:`PM_DEVICE_STATE_ACTIVE`. In case the device is physically
+suspended, the init function should call
+:c:func:`pm_device_runtime_init_suspended` before calling
+:c:func:`pm_device_runtime_enable`.
 
 .. code-block:: c
 
     /* device driver initialization function */
     static int mydev_init(const struct device *dev)
     {
+        int ret;
         ...
-        /* make sure the device physically is suspended */
+
+        /* OPTIONAL: mark device as suspended if it is physically suspended */
+        pm_device_runtime_init_suspended(dev);
+
         /* enable device runtime power management */
-        pm_device_runtime_enable(dev);
-        ...
+        ret = pm_device_runtime_enable(dev);
```

Comment:
enable / disable runtime should be done by the application and not by the driver, isn't ?

Reply:
IMO, applications should be able to decide whether runtime PM is used for a particular device or not. Regarding initialization, though, I'm not sure deferring the responsibility to the application is a good idea in all situations. For example, assuming we have a flash being used by a file system, the application may never care about the flash device. Or, an application may never care about a bus (i2c/spi) but the sensor attached to it.

If drivers can enable runtime PM, we allow the STM32 GPIO scenario: device is never clocked until used. If application is the only part that should call enable, then the init function must leave device active.

Reply:
Yep, that is what we had brainstormed, the problem in make this decision
per device is how to deal with dependencies. We may have a build option
that if it is selected, all devices supporting it have it enabled. Then
the application can disable per device if it wants, but will be application
responsability keep it consistent.

Reply:
If I got this discussion right:

I don't think application driven works well with device runtime. Technically this may be easier to handle. But from users perspective, it's complicated to explain:
"when the device runtime PM is enabled, devices are enabled only when there's a user, except at startup where devices are all enabled, and then up to the application to disable them one by one, when not required". 
This also raises the point of application genericity since additional treatment by the application would be required each time a new device is added.

I get the dependencies issue, but I don't think it is specific to this particular use case. It would raise anyway in other cases and would have to be dealt with some way or another.

Reply:
I'm affraid I have not understood the comment. The basically idea is that the application can enable / disable this feature on a device. The device itself will handle when to suspend / resume the device calling pm_device_runtim_get / put. What is the case about new a device be added ? You meant add a device at runtime ?